smart-contract-verifier/src/verifier/errors.rs

```diff
@@ -38,7 +38,7 @@ pub enum VerificationErrorKind {
     #[error("cannot parse metadata")]
     MetadataParse(String),
     #[error("compiler versions included into metadata hash does not match: {0:?}")]
-    CompilerVersionMismatch(Mismatch<Option<String>>),
+    CompilerVersionMismatch(Mismatch<Option<semver::Version>>),
```

Comment:
Can we remove Option here? I think you did it because of Option<Version> in base_verifier.rs, I propose this way:

`base_verifier.rs`
<img width="725" alt="image" src="https://user-images.githubusercontent.com/41516657/191191484-fad40ac5-8284-4c27-9449-15ac77312657.png">